bootcamp_project/Module 8/More_OOP_Concepts/VideoGame_Store_OOP_EMira1.ts

```diff
@@ -0,0 +1,79 @@
+class VideoGame {
+
+    // Static attributes
+    static availableGames: VideoGame[] = [];
+    static soldOutGames: VideoGame[] = [];
+    static totalMoneyEarned: number;
+
+
+    // Instance attributes
+    title: string;
+    developer: string;
+    availableCopies: number;
+    price: number;
+    copiesSold: number;
+
+    constructor(title: string, developer: string, availableCopies: number, price: number, copiesSold: number) {
+        this.title = title;
+        this.developer = developer;
+        this.availableCopies = availableCopies;
+        this.price = price;
+        this.copiesSold = copiesSold;
+
+        VideoGame.availableGames.push(this);
+    }
+
+    // Instance methods
+    isAvailable(): boolean {
+    return this.availableCopies > 0
+    }
+
+    sell(numberOfCopies:number = 1): number {
+
+        try {
+            if ( this.isAvailable() ) {
```

Comment:
I just noticed the availability should be defined taking into account the number of copies being sold. This means that probably the isAvailable method should receive an argument and compare the availability against it instead of comparing it to 0.